jwst/associations/lib/rules_level3_base.py

```diff
@@ -294,6 +299,50 @@ def make_member(self, item):
         }
         return member
 
+    def make_fixedslit_bkg(self):
+        """Add a background to a MIR_lrs-fixedslit observation"""
+        for product in self['products']:
+            members = product['members']
+
+            # Split out the science vs. non-science
+            # The non-science exposures will get attached
+            # to every resulting association.
+            science_exps = [
+                member
+                for member in members
+                if member['exptype'] == 'science'
+            ]
+            nonscience_exps = [
+                member
+                for member in members
+                if member['exptype'] != 'science'
+            ]
+            # Create new members for each science exposure in the association,
+            # using the the base name + _x1d as background.
+            results = []
+            new_members = self.current_product['members']
+            # Loop over all science exposures in the association
+            for science_exp in science_exps:
+                sci_name = science_exp['expname']
+                science_exp['expname'] = sci_name
+                # Construct the name for the background file
+                bkg_name = remove_suffix(
+                    splitext(split(science_exp['expname'])[1])[0])[0]
+                bkg_name = bkg_name+'_x1d.fits'
+                now_background = copy.copy(science_exp)
+                now_background['expname'] = bkg_name
+                now_background['exptype'] = 'background'
+                # Add the background file to the association table
+                new_members.append(now_background)
+
+            new_members += nonscience_exps
```

Comment:
This appears to be doubling the non-science members since they are already apart of the new_members list.

Reply:
removed.